### PR TITLE
TF2OpenAPI E2E Tests

### PR DESCRIPTION
**Pull request by jc2729**
**What this PR does / why we need it**: Add TF2OpenAPI E2E Tests

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/247)

### CODE REVIEW

File: tools/tf2openapi/cmd/main.go [hunk 1]
```diff
@@ -5,11 +5,19 @@ import (
...
 	"io/ioutil"
 	"log"
 	"os"
 )
 
+// Known error messages
+const (
+	ModelBasePathError    = "Error reading file %s \n%s"
+	OutputFilePathError = "failed writing to %s: %s"
+	SavedModelFormatError = "SavedModel not in expected format. May be corrupted: "
```
Comment (jiwang576) [inline tools/tf2openapi/cmd/main.go]: Uppercase for f in "failed"?

File: tools/tf2openapi/cmd/e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+	spec, err := cmd.Output()
+	g.Expect(err).Should(gomega.BeNil())
+	swagger := &openapi3.Swagger{}
+	g.Expect(json.Unmarshal([]byte(spec), &swagger)).To(gomega.Succeed())
+
+	expectedSpec := readFile("TestCustomFlags.golden", t)
+	expectedSwagger := &openapi3.Swagger{}
+	g.Expect(json.Unmarshal(expectedSpec, &expectedSwagger)).To(gomega.Succeed())
+
+	// test equality, ignoring order in JSON arrays
```
Comment (jiwang576) [inline tools/tf2openapi/cmd/e2e_test.go]: Could these individual Expect statements be abstracted into a `expectJsonEquality(swagger, expectedSwagger)` method?

Then this method could be reused here and also in `TestCensus` above.

Comment (jc2729) [reply tools/tf2openapi/cmd/e2e_test.go]: done

Comment (jc2729) [reply tools/tf2openapi/cmd/e2e_test.go]: done

File: tools/tf2openapi/cmd/e2e_test.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+	"path/filepath"
+	"testing"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/onsi/gomega"
+
+	"github.com/kubeflow/kfserving/tools/tf2openapi/types"
+)
+
+func TestFlowers(t *testing.T) {
```
Comment (ellis-bigelow) [inline tools/tf2openapi/cmd/e2e_test.go]: Consider templatizing these tests.

Comment (ellis-bigelow) [reply tools/tf2openapi/cmd/e2e_test.go]: i.e., try and remove all the redundant logic for each test. You're basically mapping input to outputs. Check how this is done elsewhere in the KFServing

Comment (jc2729) [reply tools/tf2openapi/cmd/e2e_test.go]: done

**Comment by ellis-bigelow** [comment]
/lgtm
/approve
!!

**Pull request merged**